tests/TestU01.hs

```diff
@@ -161,8 +161,8 @@ parseArgs (x : xs) p = do
 singleArg :: Parser a -> String -> [String] -> Either String ([String], Parser a)
 singleArg (Pure _)           x _  = Left $ "Extra argument " ++ x
 singleArg (Ap Arg p)         x xs
-    | null x || head x /= '-'     = Right (xs, fmap ($ x) p)
-    | otherwise                   = fmap2 (Ap Arg) (singleArg p x xs)
```

Comment:
The `head` function now gives a "partial function" warning, so replace it with something total.